v2/uuid/pgxcompat/uuid_test.go

```diff
@@ -0,0 +1,252 @@
+package pgxcompat_test
+
+import (
+	"bytes"
+	"reflect"
+	"testing"
+
+	"github.com/jackc/pgtype"
+	"github.com/jackc/pgtype/testutil"
+
+	"github.com/SKF/go-utility/v2/uuid"
+	"github.com/SKF/go-utility/v2/uuid/pgxcompat"
+)
+
+func TestUUIDTranscode(t *testing.T) {
+	testutil.TestSuccessfulTranscode(t, "uuid", []interface{}{
+		&pgxcompat.UUID{UUID: uuid.UUID("5b9cb067-8180-4953-846c-be5764532dc0"), Status: pgtype.Present},
+		&pgxcompat.UUID{Status: pgtype.Null},
+	})
+}
+
+type SomeUUIDWrapper struct {
+	SomeUUIDType
+}
+
+type SomeUUIDType [16]byte
+
+func TestUUIDSet(t *testing.T) {
+	successfulTests := []struct {
+		name   string
+		source interface{}
+		result pgxcompat.UUID
+	}{
+		{
+			name:   "nil",
+			source: nil,
+			result: pgxcompat.UUID{Status: pgtype.Null},
+		},
+		{
+			name:   "byte array",
+			source: [16]byte{0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15},
+			result: pgxcompat.UUID{UUID: uuid.UUID("00010203-0405-0607-0809-0a0b0c0d0e0f"), Status: pgtype.Present},
+		},
+		{
+			name:   "byte slice",
+			source: []byte{0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15},
+			result: pgxcompat.UUID{UUID: uuid.UUID("00010203-0405-0607-0809-0a0b0c0d0e0f"), Status: pgtype.Present},
+		},
+		{
+			name:   "nil byte slice",
+			source: ([]byte)(nil),
+			result: pgxcompat.UUID{Status: pgtype.Null},
+		},
+		{
+			name:   "string",
+			source: "00010203-0405-0607-0809-0a0b0c0d0e0f",
+			result: pgxcompat.UUID{UUID: uuid.UUID("00010203-0405-0607-0809-0a0b0c0d0e0f"), Status: pgtype.Present},
+		},
+		{
+			name:   "string without dashes",
+			source: "000102030405060708090a0b0c0d0e0f",
+			result: pgxcompat.UUID{UUID: uuid.UUID("00010203-0405-0607-0809-0a0b0c0d0e0f"), Status: pgtype.Present},
+		},
+	}
+
+	for _, tt := range successfulTests {
+		var r pgxcompat.UUID
+
+		err := r.Set(tt.source)
+		if err != nil {
+			t.Errorf("%s: %v", tt.name, err)
+		}
+
+		if r != tt.result {
+			t.Errorf("%s: expected %v to convert to %v, but it was %v", tt.name, tt.source, tt.result, r)
+		}
+	}
+}
+
+func TestUUIDAssignTo(t *testing.T) { // nolint:gocyclo
+	{
```

Comment:
This is new for me. Why all these curly brackets?

Reply:
Each test gets a clean scope, so both `dst` and `expected` can have different types and we can avoid having `stingDst`, `someUUIDTypeDst` etc. It is mostly copied from here: https://github.com/jackc/pgtype/blob/master/uuid_test.go but I find it rather nice :)

Reply:
Makes sense, I'll create separate test funcs